//
// PROJECT:         Aspia
// FILE:            category/category_ras.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message RAS
{
    enum FramingProtocol
    {
        FRAMING_PROTOCOL_UNKNOWN = 0;
        FRAMING_PROTOCOL_PPP     = 1;
        FRAMING_PROTOCOL_SLIP    = 2;
    }

    enum Options
    {
        OPTION_UNKNOWN                       = 0;
        OPTION_IP_HEADER_COMPRESSION         = 1;
        OPTION_PPP_LCP_EXTENSIONS            = 2;
        OPTION_SW_COMPRESSION                = 4;
        OPTION_REQUIRE_ENCRYPTED_PASSWORD    = 8;
        OPTION_REQUIRE_MS_ENCRYPTED_PASSWORD = 16;
        OPTION_REQUIRE_DATA_ENCRYPTION       = 32;
        OPTION_REQUIRE_EAP                   = 64;
        OPTION_REQUIRE_PAP                   = 128;
        OPTION_REQUIRE_SPAP                  = 256;
        OPTION_REQUIRE_CHAP                  = 512;
        OPTION_REQUIRE_MS_CHAP               = 1024;
        OPTION_REQUIRE_MS_CHAP_2             = 2048;
    }

    message Item
    {
        string connection_name           = 1;
        string device_name               = 2;
        string device_type               = 3;
        uint32 country_code              = 4;
        string area_code                 = 5;
        string local_phone_number        = 6;
        string user_name                 = 7;
        string domain                    = 8;
        string ip_address                = 9;
        string dns_address_1             = 10;
        string dns_address_2             = 11;
        string wins_address_1            = 12;
        string wins_address_2            = 13;
        FramingProtocol framing_protocol = 14;
        string script                    = 15;
        uint32 options                   = 16;
    }

    repeated Item item = 1;
}
